homeassistant/components/media_player/onkyo_serial.py

```diff
@@ -0,0 +1,323 @@
+"""
+Support for Onkyo Receivers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.onkyoserial/
+"""
+import logging
+import threading
+import re
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_SELECT_SOURCE, SUPPORT_PLAY, MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (STATE_OFF, STATE_ON, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = []

```

Comment:
You can make a mini library only for hass. `denon.py` have the logic on other side of telnet and use 10~ lines. REST are a grey area. Only simple calls are allow -> if I use `lib.do_stuff` or `request.get('/do/stuff')` is in the end the same and the logic is outside hass and the protocol is in request library. But I think also the iTunes are to big for include hass, but that is allready merge but I give a hint to Autor.